// Copyright 2022 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto2";

option optimize_for = LITE_RUNTIME;

package web_app.proto;

// Store size and timestamp of an icon so that changes can be easily detected.
message ShortcutIconData {
  optional int32 icon_size = 1;
  optional int64 timestamp = 2;
}

message ShortcutState {
  optional string title = 1;
  optional string description = 2;
  repeated ShortcutIconData icon_data_any = 3;
}

message WebAppProtocolHandler {
  optional string protocol = 1;
  optional string url = 2;
}

// Represents all the common OS Integration states to be stored in the web_app
// DB that matches the values in the OS.
message WebAppOsIntegrationState {
  optional ShortcutState shortcut_states = 1;
  repeated WebAppProtocolHandler manifest_protocol_handlers_states = 2;
  // Add data states for other OS integration hooks here.
}